src/rogue/interfaces/memory/Block.cpp

```diff
@@ -190,21 +192,28 @@ void rim::Block::intStartTransaction(uint32_t type, bool forceWr, bool check, ri
          highByte = size_-1;
          if ( type == rim::Write || type == rim::Post ) {
             stale_ = false;
-            for ( vit = variables_.begin(); vit != variables_.end(); ++vit ) (*vit)->stale_ = false;
+            for ( vit = variables_.begin(); vit != variables_.end(); ++vit ) {
+              (*vit)->stale_ = false;
+            }
          }
       } else {
 
-          if (index < 0 || index >= var->numValues_) {
-              lowByte = var->lowTranByte_;
-              highByte = var->highTranByte_;
-          }
-
-          else {
-              lowByte = var->listLowTranByte_[index];
-              highByte = var->listHighTranByte_[index];
-          }
+        if ( type == rim::Read || type == rim::Verify ) {
+            if (index < 0 || index >= var->numValues_) {
+                 lowByte = var->lowTranByte_;
+                 highByte = var->highTranByte_;
+             } else {
+                 lowByte = var->listLowTranByte_[index];
+                 highByte = var->listHighTranByte_[index];
+             }
+         }
+         else {
 
-          if ( type == rim::Write || type == rim::Post ) {
+             lowByte = var->staleLowByte_;
+             highByte = var->staleHighByte_;
+             // Catch case where fewer stale bytes than min access or non-aligned
+             if (lowByte % minAccess != 0) lowByte -= lowByte % minAccess;
+             if ((highByte+1) % minAccess != 0) highByte += minAccess - ((highByte+1) % minAccess);
```

Comment:
@slacrherbst This part here feels like a hack. Maybe `staleLowByte_` and `staleHighByte_` should obey the min access requirements at the time they are marked?